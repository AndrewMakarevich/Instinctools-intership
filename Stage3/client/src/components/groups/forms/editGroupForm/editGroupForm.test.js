import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithRouter } from '../../../../test/helpers/renderWith';
import EditGroupForm from './editGroupForm';

describe("Edit group form's correct", () => {
  test('render', () => {
    renderWithRouter(
      <EditGroupForm
        groupObj={{ groupName: 'FirstGroup', groupTitle: 'First' }}
      />
    );
    expect(screen.getByTestId('edit-form')).toBeInTheDocument();
    expect(screen.getAllByTestId('my-label-input').length).toBe(2);
    expect(screen.getByText('Clear changes')).toBeInTheDocument();
    expect(screen.getByTestId('submit-group-changes-btn')).toBeInTheDocument();
  });
});
