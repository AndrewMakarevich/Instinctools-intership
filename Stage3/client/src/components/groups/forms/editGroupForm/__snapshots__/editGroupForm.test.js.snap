// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit group form's correct render 1`] = `
<form
  class="form"
  data-testid="edit-group-form"
>
  <div
    class="form-inputs__wrapper"
  >
    <label
      class="label "
      data-testid="my-label-input"
    >
      <input
        class="my-input "
        placeholder=" "
        value="FirstGroup"
      />
      <span>
        Group's groupName
      </span>
    </label>
    <label
      class="label "
      data-testid="my-label-input"
    >
      <input
        class="my-input "
        placeholder=" "
        value="First"
      />
      <span>
        Group's groupTitle
      </span>
    </label>
  </div>
  <div
    class="form-buttons__wrapper"
  >
    <button
      class="my-button "
    >
      Clear changes
    </button>
    <button
      class="my-button "
      data-testid="submit-group-changes-btn"
    >
      Submit changes
    </button>
  </div>
</form>
`;
